de.metas.contracts/src/main/java/de/metas/flatrate/process/C_Flatrate_Term_Create.java

```diff
@@ -122,24 +126,45 @@ private I_C_Flatrate_Term createTerm(final I_C_BPartner partner)
 	{
 		final IContextAware context = PlainContextAware.newWithThreadInheritedTrx(getCtx());
 		final boolean completeIt = true;
-		final I_C_Flatrate_Term term = flatrateBL.createTerm(context, partner, conditions, startDate, userInCharge, product, completeIt);
-		if (term == null)
+
+		final I_C_Flatrate_Term newTerm = flatrateBL.createTerm(context, partner, conditions, startDate, userInCharge, product, false);
+
+		if (newTerm == null)
 		{
 			return null;
 		}
 
 		if (product != null)
 		{
-			term.setM_Product_ID(product.getM_Product_ID());
+			newTerm.setM_Product_ID(product.getM_Product_ID());
 		}
 
 		if (endDate != null)
 		{
-			term.setEndDate(endDate);
+			newTerm.setEndDate(endDate);
+		}
+
+		InterfaceWrapperHelper.save(newTerm);
+
+		// task #1169
+		// in case the term to be created will overlap with other term regarding time period and product
+		// the user must be announced about this and the new term shall not be completed
+		final boolean hasOverlappingTerms = flatrateBL.hasOverlappingTerms(newTerm);
```

Comment:
Please make this check not in this process, but in a model interceptor.
Goal: the check shall also be made if a term is completed without this process taking place